Python/ceval.c

```diff
@@ -4744,8 +4744,10 @@ PyEval_EvalCodeEx(PyObject *_co, PyObject *globals, PyObject *locals,
         Py_DECREF(defaults);
         return NULL;
     }
+#ifdef Py_DEBUG
```

Comment:
For assert, you must rely on NDEBUG macro, not on Py_DEBUG. It's possible to build Python in release mode with assertions for example (we got bug reports about that).

```suggestion
#ifndef NDEBUG
```

Alternative using a long line: `assert ((((PyCodeObject *)_co)->co_flags & (CO_NEWLOCALS | CO_OPTIMIZED)) == 0);`.